gluten-substrait/src/main/scala/org/apache/spark/sql/execution/ColumnarSubqueryBroadcastExec.scala

```diff
@@ -64,6 +65,14 @@ case class ColumnarSubqueryBroadcastExec(
     copy(name = "native-dpp", buildKeys = keys, child = child.canonicalized)
   }
 
+  // Copy from org.apache.spark.sql.execution.joins.HashJoin#canRewriteAsLongType
+  // we should keep consistent with it to identify the LongHashRelation.
+  private def canRewriteAsLongType(keys: Seq[Expression]): Boolean = {
```

Comment:
@yikf, if spark changes this logic, e.g., for adding support for other types, unexpected failure can happen in Gluten? Seems there is no other way.

Reply:
There doesn't seem to be any other way at the moment.

Reply:
@yikf, ok, let's merge this pr firstly. Thanks for your efforts!

Reply:
thanks @PHILO-HE